### PR TITLE
Feat/vue3 migration

### PR DESCRIPTION
**Pull request by soulsam480**
There are many things to be refactored for this APP to be fully compatible with vue 3. Currently I've refactored the bare minimum. When I run the dev server I see some error in the dev preview. Can anybody pull this PR and check ? I'm also unable to start the electron dev server.

#### Considerations:
- Initially planned to use @vitejs but decided to stick with vue-cli as this APP is heavily reliant on vue-cli webpack related packages
- As the landing component is huge 2000+ LOC, I think there is a lot of room for refactor and I need help of the maintainers to do so
- I also plan to convert the UI components to the recommended <script setup> syntax
- A complete TS rewrite is also planned (not now as the Landing component is huge)

Again thanks, will update this PR

### CODE REVIEW

File: .pnpm-debug.log [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (Stanzilla) [inline .pnpm-debug.log:1]: file should be in `.gitignore`

File: .prettierrc.json [hunk 2]
```diff
@@ -1,3 +1,9 @@
 {
-    "endOfLine": "auto"
-}
\ No newline at end of file
+  "semi": true,
+  "arrowParens": "always",
+  "printWidth": 120,
+  "bracketSpacing": true,
+  "trailingComma": "all",
+  "singleQuote": true,
```
Comment (Stanzilla) [inline .prettierrc.json:7]: would actually prefer leaving that at double quotes to avoid too many unnecessary changes in the PR

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -0,0 +1,17779 @@
+lockfileVersion: 5.3
```
Comment (Stanzilla) [inline pnpm-lock.yaml]: Should also be in gitignore since we' re using npm not yarn

**Comment by Stanzilla** [comment]
When you move files from js to ts and also make changes to it, can you make the move in a separate commit so git tracks it?

**Comment by mrbuds** [comment]
Could you move indentation changes without code logic modification to a separate commit?

File: .prettierrc.json (hunk 2 above)
Comment (soulsam480) [reply .prettierrc.json:7]: I'm also thinking the same to revert prettier to previous config. That will remove too many diffs

File: pnpm-lock.yaml (hunk 3 above)
Comment (soulsam480) [reply pnpm-lock.yaml]: yes these will be removed

File: .pnpm-debug.log (hunk 1 above)
Comment (soulsam480) [reply .pnpm-debug.log:1]: yes Will remove it in next commit and add it to .gitignore

**Comment by Stanzilla** [comment]
I cleaned this up a bit and made sure it compiles again

**Comment by Stanzilla** [comment]
TODO https://github.com/nklayman/vue-cli-plugin-electron-builder/releases/tag/v3.0.0-alpha.0

**Comment by Causese** [comment]
3 errors below that are related. Can reproduce by following video steps identially but it's awful
https://www.youtube.com/watch?v=LzbwBsFpq9M

Tried to reproduce from 1:57min and onwards and it didn't work so i guess some earlier steps play a role

test

```
1x WeakAuras\RegionTypes\Group-Group.lua:102: WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua:5208: attempt to index local 'data' (a nil value)
[string "=[C]"]: in function `(for generator)'
[string "@WeakAuras\RegionTypes\Group-Group.lua"]:102: in function `modify'
[string "@WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua"]:2773: in function `SetRegion'
[string "@WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua"]:2591: in function <WeakAuras\WeakAuras.lua:2550>
[string "@WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua"]:2695: in function `Add'
[string "@WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua"]:2703: in function `AddParents'
[string "@WeakAuras\WeakAuras-3.7.3-59-g9a88083.lua"]:1868: in function `Delete'
[string "@WeakAurasOptions\WeakAurasOptions-3.7.3-59-g9a88083.lua"]:447: in function `OnAccept'
[string "@FrameXML\StaticPopup.lua"]:5148: in function `StaticPopup_OnClick'
[string "*:OnClick"]:1: in function <[string "*:OnClick"]:1>

Locals:
 = <no value>
```

```
2x ...eGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua:966: attempt to index local 'childData' (a nil value)
[string "@WeakAurasOptions\AceGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua"]:966: in function `addChildrenNames'
[string "@WeakAurasOptions\AceGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua"]:971: in function `SetNormalTooltip'
[string "@WeakAurasOptions\AceGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua"]:804: in function <...eGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua:799>

Locals:
data = <table> {
 controlledChildren = <table> {
 }
 borderBackdrop = "Blizzard Tooltip"
 authorOptions = <table> {
 }
 preferToUpdate = false
 yOffset = 0
 anchorPoint = "CENTER"
 borderColor = <table> {
 }
 actions = <table> {
 }
 triggers = <table> {
 }
 internalVersion = 48
 selfPoint = "CENTER"
 subRegions = <table> {
 }
 load = <table> {
 }
 backdropColor = <table> {
 }
 source = "import"
 scale = 1
 border = false
 borderEdge = "Square Full White"
 regionType = "group"
 borderSize = 2
 borderOffset = 4
 tocversion = 90200
 id = "New 10"
 frameStrata = 1
 anchorFrameType = "SCREEN"
 animation = <table> {
 }
 borderInset = 1
 xOffset = 0
 uid = "S667pc4F2Nx"
 conditions = <table> {
 }
 information = <table> {
 }
 config = <table> {
 }
}
indent = "  "
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = "New"
 2 = "New 4"
 3 = "New 4 2"
 4 = "New 2 2"
}
(for control) = 1
index = 1
childId = "New"
childData = nil
(*temporary) = "New"
(*temporary) = nil
(*temporary) = "New"
(*temporary) = nil
(*temporary) = "attempt to index local 'childData' (a nil value)"
pairs = <function> defined =[C]:-1
tinsert = <function> defined =[C]:-1
namestable = <table> {
 1 = ""
 2 = "  New"
}
addChildrenNames = <function> defined @WeakAurasOptions\AceGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua:962
```

```
1x ...faceWeakAurasOptions\RegionOptions\Group-Group.lua:594: attempt to index local 'childData' (a nil value)
[string "@WeakAurasOptions\RegionOptions\Group-Group.lua"]:594: in function `create'
[string "@WeakAurasOptions\GroupOptions.lua"]:13: in function `?'
[string "@WeakAurasOptions\OptionsFrames\OptionsFrame.lua"]:963: in function `EnsureOptions'
[string "@WeakAurasOptions\OptionsFrames\OptionsFrame.lua"]:940: in function `UpdateOptions'
[string "@WeakAurasOptions\OptionsFrames\OptionsFrame.lua"]:982: in function `FillOptions'
[string "@WeakAurasOptions\OptionsFrames\OptionsFrame.lua"]:1303: in function `PickDisplay'
[string "@WeakAurasOptions\WeakAurasOptions-3.7.3-59-g9a88083.lua"]:1191: in function `PickDisplay'
[string "@WeakAurasOptions\AceGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua"]:499: in function <...eGUI-Widgets\AceGUIWidget-WeakAurasDisplayButton.lua:464>

Locals:
id = "New 10"
data = <table> {
 controlledChildren = <table> {
 }
 borderBackdrop = "Blizzard Tooltip"
 authorOptions = <table> {
 }
 preferToUpdate = false
 yOffset = 0
 anchorPoint = "CENTER"
 borderColor = <table> {
 }
 actions = <table> {
 }
 triggers = <table> {
 }
 internalVersion = 48
 selfPoint = "CENTER"
 subRegions = <table> {
 }
 load = <table> {
 }
 backdropColor = <table> {
 }
 source = "import"
 scale = 1
 border = false
 borderEdge = "Square Full White"
 regionType = "group"
 borderSize = 2
 borderOffset = 4
 tocversion = 90200
 id = "New 10"
 frameStrata = 1
 anchorFrameType = "SCREEN"
 animation = <table> {
 }
 borderInset = 1
 xOffset = 0
 uid = "S667pc4F2Nx"
 conditions = <table> {
 }
 information = <table> {
 }
 config = <table> {
 }
}
options = <table> {
 __title = "Group Settings"
 chooseIcon = <table> {
 }
 __order = 1
 endHeader = <table> {
 }
 scale = <table> {
 }
 groupIcon = <table> {
 }
}
hasSubGroups = false
hasDynamicSubGroup = false
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = "New"
 2 = "New 4"
 3 = "New 4 2"
 4 = "New 2 2"
}
(for control) = 1
index = 1
childId = "New"
childData = nil
(*temporary) = "New"
(*temporary) = nil
(*temporary) = "attempt to index local 'childData' (a nil value)"
L = <table> {
 No Instance = "No Instance"
 Scenario = "Scenario"
 Destination Reaction = "Destination Reaction"
 Threat Value = "Threat Value"
 A 48x48 pixels icon = "A 48x48 pixels icon"
 Stacks = "Stacks"
 Aura(s) Missing = "Aura(s) Missing"
 Incoming Heal = "Incoming Heal"
 Armor against Target (%) = "Armor against Target (%)"
 Show Heal Absorb = "Show Heal Absorb"
 Frost Resistance = "Frost Resistance"
 Aggro = "Aggro"
 Alpha = "Alpha"
 Player Character = "Player Character"
 If unchecked, then a default color will be used (usually yellow) = "If unchecked, then a default color will be used (usually yellow)"
 Down = "Down"
 Always active trigger = "Always active trigger"
 In Vehicle = "In Vehicle"
 Down, then Right = "Down, then Right"
 Shows a custom texture = "Shows a custom texture"
 Resurrect = "Resurrect"
 Rotate Right = "Rotate Right"
 Pulse = "Pulse"
 Fail Alert = "Fail Alert"
 Role = "Role"
 Vertical Bar = "Vertical Bar"
 Angle = "Angle"
 Not On Threat Table = "Not On Threat Table"
 Hit (%) = "Hit (%)"
 25 Player Raid (Normal) = "25 Player Raid (Normal)"
 Always include realm = "Always include realm"
 Glow %s = "Glow %s"
 Y Offset = "Y Offset"
 Loop = "Loop"
 Item Slot String = "Item Slot String"
 Frequency = "Frequency"
 Source = "Source"
 Animations = "Animations"
 Large = "Large"
 Can be a name or a UID (e.g., party1). A name only works on friendly players in your group. = "Can be a name or a UID (e.g., party1). A name only works on friendly players in your group."
 Multiple Triggers = "Multiple Triggers"
 %s - Custom Sort = "%s - Custom Sort"
 Cast Bar = "Cast Bar"
 Custom Sort = "Custom Sort"
 On Hide = "On Hide"
 Current Movement Speed (%) = "Current Movement Speed (%)"
 Small Icon = "Small Icon"
 The following auras will be added: %s
 = "The following auras will be added: %s
"
 Block against Target (%) = "Block against Target (%)"
 Add Missing Auras = "Add Missing Auras"
 Automatic Icon = "Automatic Icon"
 Overlay Latency = "Overlay Latency"
 Contains = "Contains"
 Inherited = "Inherited"
 Player Effective Level = "Player Effective Level"
 Prevents duration information from decreasing when an aura refreshes. May cause problems if used with multiple auras with different durations. = "Prevents duration information from decreasing when an aura refreshes. May cause problems if used with multiple auras with different durations."
 Column Space = "Column Space"
 Parry (%) = "Parry (%)"
 Extra Attacks = "Extra Attacks"
 Timed = "Timed"
 Ungroup = "Ungroup"
 Remaining Duration = "Remaining Duration"
 Negator = "Not"
 Cooldown Settings = "Cooldown Settings"
 Raid (Timewalking) = "Raid (Timewalking)"
 Only if selected = "Only if selected"
 5 Man Dungeon = "5 Man Dungeon"
 Icon Inset = "Icon Inset"
 Player Faction = "Player Faction"
 Font Type = "Font Type"
 Print Profi
```

**Comment by mrbuds** [comment]
Vue3 migration was merged from an other branch
I replaced a few libs, dropped toasts as there is no good replacement, switched to Pinata for config store
Project is a bit more healthy for future changes now
Thanks for your contribution!
